apps/src/p5lab/AnimationPicker/AnimationPickerBody.jsx

```diff
@@ -52,26 +53,80 @@ class AnimationPickerBody extends React.Component {
     spriteLab: PropTypes.bool
   };
 
-  state = {
-    searchQuery: '',
-    categoryQuery: '',
-    currentPage: 0
+  constructor(props) {
+    super(props);
+
+    const initialState = {
+      searchQuery: '',
+      categoryQuery: '',
+      currentPage: 0,
+      libraryManifest: props.spriteLab ? spriteCostumeLibrary : animationLibrary
+    };
+    const results = this.searchAssets(initialState.currentPage, initialState);
+    this.state = {
+      ...initialState,
+      results
+    };
+  }
+
+  searchAssets = (page, config = {}) => {
+    let {searchQuery, categoryQuery, libraryManifest} = config;
+
+    // Set defaults if any config values are undefined.
+    if (searchQuery === undefined) {
+      searchQuery = this.state.searchQuery;
+    }
+    if (categoryQuery === undefined) {
+      categoryQuery = this.state.categoryQuery;
+    }
+    if (libraryManifest === undefined) {
+      libraryManifest = this.state.libraryManifest;
+    }
+
+    const {results} = searchAssets(
+      searchQuery,
+      categoryQuery,
+      libraryManifest,
+      page,
+      MAX_SEARCH_RESULTS
+    );
+    return results;
   };
 
-  onSearchQueryChange = value => {
-    this.setState({searchQuery: value, currentPage: 0});
+  handleScroll = event => {
```

Comment:
NIT: Since you access the state variable more than once, consider using this pattern:
`const {currentPage, results} = this.state`